plyer/platforms/macosx/maps.py

```diff
@@ -0,0 +1,90 @@
+'''
+Module of MacOSX API for plyer.maps.
+'''
+
+from subprocess import Popen, PIPE
+from plyer.facades import Maps
+from urllib.parse import quote_plus
+
+
+class MacOSMaps(Maps):
+    '''
+    Implementation of MacOS Maps API.
+    '''
+
+    def _open_by_address(self, address, **kwargs):
+        '''
+        :param address: An address string that geolocation can understand.
+        '''
+
+        address = quote_plus(address, safe=',')
+        maps_address = 'http://maps.apple.com/?address=' + address
+
+        process = Popen(
+            ['open', '-a', 'Maps', maps_address],
+            stdout=PIPE, stderr=PIPE)
+        stdout, stderr = process.communicate()
+
+    def _open_by_lat_long(self, latitude, longitude, **kwargs):
+        '''
+        Open a coordinate span denoting a latitudinal delta and a
+        longitudinal delta (similar to MKCoordinateSpan)
+
+        :param name: (optional), will set the name of the dropped pin
+        '''
+
+        name = kwargs.get("name", "Selected Location")
+        maps_address = 'http://maps.apple.com/?ll={},{}&q={}'.format(
+            latitude, longitude, name)
+
+        process = Popen(
+            ['open', '-a', 'Maps', maps_address],
+            stdout=PIPE, stderr=PIPE)
+        stdout, stderr = process.communicate()
+
+    def _search(self, query, **kwargs):
+        '''
+        :param query: A string that describes the search object (ex. "Pizza")
+
+        :param latitude: (optional), narrow down query within area,
+        MUST BE USED WITH LONGITUDE
+
+        :param longitude: (optional), narrow down query within area,
+        MUST BE USED WITH LATITUDE
+        '''
+
+        latitude = kwargs.get('latitude')
+        longitude = kwargs.get('longitude')
+
+        query = quote_plus(query, safe=',')
+        maps_address = 'http://maps.apple.com/?q=' + query
+
+        if latitude is not None and longitude is not None:
+            maps_address += '&sll={},{}'.format(latitude, longitude)
+
+        process = Popen(
+            ['open', '-a', 'Maps', maps_address],
+            stdout=PIPE, stderr=PIPE)
+        stdout, stderr = process.communicate()
```

Comment:
What about:
```python
import webbrowser
webbrowser.open("http://maps.apple.com/?q=Tour Eiffel")
```

That will also be likely (did not tested yet, but should work) portable to iOS.

Reply:
I did consider this but by this route there is a dialogue that pops up every time asking "Would you like to open this in Maps" while my current method specifies it via the Maps app and it opens w/o any dialogue. 

All in all, minor but I personally prefer this as it's cleaner and more direct. Let me know your thoughts on this @misl6

Reply:
> I did consider this but by this route there is a dialogue that pops up every time asking "Would you like to open this in Maps" while my current method specifies it via the Maps app and it opens w/o any dialogue.

Not happening on my side, but it may be a specific configuration / "Don't ask anymore" flag.

`open -a Maps` is not documented anywhere, and this is why I'm worried about future potential headaches.
But, everything can change (even the link structure), so feel free to use your implementation, if you're fine with it 😀